core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -321,7 +321,7 @@ private[spark] class Executor(
           logInfo(s"Executor killed $taskName (TID $taskId)")
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled))
 
-        case cDE: CommitDeniedException =>
+        case CausedBy(cDE: CommitDeniedException) =>
```

Comment:
Oh I missed that, nice one. Yeah I agree this is a good one then, certainly if it can replace similar patterns elsewhere.